app/components/button/index.tsx

```diff
@@ -48,12 +58,22 @@ const Button = ({
             onPress={onPress}
             testID={testID}
         >
-            <Text
-                style={txtStyle}
-                numberOfLines={1}
-            >
-                {text}
-            </Text>
+            <View style={styles.container}>
+                {Boolean(iconName) &&
+                <CompassIcon
+                    name={iconName!}
+                    size={iconSize}
+                    color={theme.buttonColor}
```

Comment:
The color will probably depend on the button color (which depends on the type of button (primary, secondary, disabled, etc...). In order to do this the "correct" way, I would do something like:

```
    const txtStyle = useMemo(() => StyleSheet.flatten([
        buttonTextStyle(theme, size, emphasis, buttonType),
        textStyle,
    ]), [theme, textStyle, size, emphasis, buttonType]);
...
   <CompassIcon
       color={txtStyle.color}
...
```